### PR TITLE
mesa-lib: bump to 11.2.2. Changing up the logic for wayland, as it wa…

### PR DESCRIPTION
**Pull request by stumbles**
…s wayland was not

being added to the egl platforms. First tried fixing the spacing issue with += in the
if logic, no go. Then tried making the OPTS in the if logic to be --with-egl-platforms=drm,x11,wayland,
no go. So let's do it this way and look for it being installed.

### CODE REVIEW

**Comment by lunar-linux-b01** [comment]
SUCCESS!

File: lib/mesa-lib/BUILD [hunk 1]
```diff
@@ -9,10 +9,10 @@ else
   OPTS+=" --enable-egl --enable-osmesa --with-gallium-drivers="
 fi  &&
 
-OPTS+=" --with-egl-platforms=drm,x11" &&
-
-if in_depends $MODULE wayland; then
-  OPTS+=",wayland"
+if module_installed wayland; then
```
Comment (Ratler) [inline lib/mesa-lib/BUILD]: @stumbles using module_installed wayland is flawed, please don't do that. This does not make it optional for the user when wayland is installed. You can use your proposed logic though, just change that "if" to if in_depends $MODULE wayland, then keep the rest of your changes :)

**Comment by stumbles** [comment]
retest this please

**Comment by lunar-linux-b01** [comment]
SUCCESS!

**Comment by Florin65** [comment]
agree with Ratler. it's better to iuse in_depends.

**Comment by stumbles** [comment]
retest this please

**Comment by lunar-linux-b01** [comment]
SUCCESS!

**Pull request merged**